src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs

```diff
@@ -442,43 +460,39 @@ private static bool CheckIfFileSystemIsCaseInsensitive()
                 var isCaseInsensitive = true;
                 foreach (var path in listOfPathsToCheck)
                 {
-                    bool ignore;
-                    var result = CheckCaseSenstivityRecursivelyTillDirectoryExists(path, out ignore);
-                    if (!ignore)
+                    string? firstParentDirectory = GetFirstParentDirectoryThatExists(path);
+                    if (firstParentDirectory is not null)
                     {
-                        isCaseInsensitive &= result;
+                        isCaseInsensitive &= CheckIfFileSystemIsCaseInsensitive(firstParentDirectory);
```

Comment:
>isCaseInsensitive &= CheckIfFileSystemIsCaseInsensitive(firstParentDirectory); [](http://example.com/codeflow?start=24&length=78)

Just for my own learning, why wouldn't it be sufficient to check any one of these paths? I.e. why not return immediately?

Reply:
I don't know.

Technically, the whole thing is flawed, because individual filesystems can be case sensitive or not. On Windows I can configure a specific partition to be case sensitive, even though Windows' default is case insensitive filesystems. I expect there are filesystems on Linux where you can enable insensitivity.  But I imagine that the number of people who change the defaults is very close to zero, and NuGet wouldn't be the only software that breaks when those defaults are changed.